as11fixity.py

```diff
@@ -33,43 +39,96 @@ def create_csv(csv_file, *args):
     print "CSV file already exists."
     
     
-    
-    
-elif checkfile == False:
+if checkfile == False:
     print "No CSV file exists."
 #3    
 
 for dirpath, dirnames, filenames in os.walk(starting_dir):
     for filename in [f for f in filenames if f.endswith(".mxf")]:
-        print os.path.join(dirpath, filename)
-#7
+        full_path = os.path.join(dirpath, filename)
+        print dirpath +  ' ---DIR PATH'
+        print filename + ' ---THIS IS FILEAME'
+        print full_path + ' ---THIS IS THE FULL_PATH'
+        #7
+        
+        file_no_path = os.path.basename(full_path)
+        print file_no_path + ' ---THIS IS FILE_NO_PATH'
+        #8.1
+        
+        file_no_extension = os.path.splitext(os.path.basename(file_no_path))[0]
+        print file_no_extension + ' ---THIS IS FILE_NO_EXTENSION'
+        #8.2
+        
+        xml_file = file_no_extension + '.xml'
+        print xml_file + '  ---This is xml file'
+        full_xml_path = os.path.join(dirpath,xml_file)
+        print full_xml_path + ' ---This id the full xml path'
+        checkfile = os.path.isfile(os.path.join(dirpath,xml_file))
+        if checkfile == True:
+            print 'XML file already exists.'
+        if checkfile == False:
+            print 'No XML file exists.'
+        #8.3
+        
+        dpp_xml_parse = etree.parse(full_xml_path)
+        print dpp_xml_parse, 'hahaha'
+        dpp_xml_namespace = dpp_xml_parse.xpath('namespace-uri(.)')
+        print dpp_xml_namespace, 'lol'
+        checksum = dpp_xml_parse.findtext('//ns:MediaChecksumValue', namespaces={'ns':dpp_xml_namespace })
+        print checksum
+        #12 
+     
+        
+        
+def md5(full_path):
+    hash_md5 = hashlib.md5()
+    with open(full_path, "rb") as f:
+        for chunk in iter(lambda: f.read(4096), b""):
+            hash_md5.update(chunk)
+    return hash_md5.hexdigest()
+print md5(full_path)
+#13
+
+if md5(full_path)==checksum:
```

Comment:
this is running it again. Instead, if you store the mxf md5 as a variable once, then you can runn the comparison on it after.
SOmething like
mxf_md5 = md5(full_path)
```
if mxf_md5 == checksum:
    print 'ALL IS WELL'
else:
    print 'OH JESUS'
```